javascript/extractor/src/com/semmle/js/extractor/FileExtractor.java

```diff
@@ -120,6 +120,12 @@ protected boolean contains(File f, String lcExt, ExtractorConfig config) {
         if (isBinaryFile(f, lcExt, config)) {
           return false;
         }
+        // for ERB files we are only interrested in `.html.erb` files
+        if (FileUtil.extension(f).equalsIgnoreCase(".erb")) {
+          if (!f.getName().endsWith(".html.erb")) {
```

Comment:
You could simplify this with
```js
if(f.getName().endsWith(".html.erb"))
  return true;
```

This code doesn't handle upper case - is this important?

Reply:
Your suggestion wouldn't work, because it would let through all `.erb` files (the `super.contains(f, lcExt, config);` call returns `true` for any `.erb` file).  

> This code doesn't handle upper case - is this important? 

I don't think so, but the existing code handles upper case, so I wanted to keep that.  

I'll keep the code as is.

Reply:
I don't think handling upper case is important, but I feel we should test things consistently. Having `equalsIgnoreCase` in combination with a case sensitive  `endsWith` feels wrong.

Reply:
Agreed on the logic - didn't see that `.erb` would have been matched in the following lines.

You could just shove in `f.getName().toLowerCase().endsWith(".html.erb")`

Reply:
Agree that we should do `f.getName().toLowerCase()` in the second branch - keeps it more obvious and also handles upper case extensions.